docs-ref-conceptual/use-cli-effectively.md

```diff
@@ -277,6 +277,38 @@ The following are Azure CLI environment variables:
 | **AZURE_CONFIG_DIR**           | The global directory for configuration files, logs, and telemetry. If unspecified, this value defaults to `~/.azure.` |
 | **AZURE_EXTENSION_DIR**        | The directory containing extension installations. If unspecified, this value defaults to the `cliextensions` subdirectory within the global configuration directory. |
 
+## Error handling for Azure CLI in PowerShell
+
+You can run Azure CLI commands in PowerShell, as described in [Choose the right Azure command-line tool](choose-the-right-azure-command-line-tool.md). If you do, be sure you understand Azure CLI error handling in PowerShell. In particular, you can't use the `try` and `catch` keywords.
+
+An alternative is to use the `$?` automatic variable. This variable contains the status of the most recent command. If the previous command fails, `$?` has the value of `$False`. For more information, see [about_Automatic_Variables](/powershell/module/microsoft.powershell.core/about/about_automatic_variables).
+
+The follow example shows how this automatic variable can work for error handling:
+
+```powershell
+az group create --name MyResourceGroup 
+if ($? -eq $False) 
+{
+Write-Error "Error creating storage account"
+}
+```
+
+The first command lacks the required `--location` parameter, so it fails. The conditional finds that `$?` is false and writes an error.
+
+By default, PowerShell catches only terminating errors. You can change this behavior by using the `$ErrorActionPreference` global variable.
+
+```powershell
+$ErrorActionPreference = "Stop"
+```
+
+After testing, be sure to change the value back:
+
+```powershell
+$ErrorActionPreference = "Continue"
+```
+
+For more information about PowerShell error handling, see [Everything you wanted to know about exceptions](/powershell/scripting/learn/deep-dives/everything-about-exceptions).
```

Comment:
I am not sure about the relevance of this information. What does this have to do with handling Azure CLI errors? These settings only affect PowerShell errors. The `Write-Error` cmdlet does not create a terminating error that would be caught by setting the preference to `Stop`.

If you are using the `try/catch` scenario I showed in the previous comment, then you need to set the preference to `Stop` so that PowerShell can handle the error.